tests/integration/conftest.py

```diff
@@ -192,6 +192,40 @@ async def cloud_type(ops_test: OpsTest):
     return cloud.cloud.type_
 
 
+async def cloud_proxied(ops_test: OpsTest):
+    """Setup a cloud proxy settings if necessary
+
+    Test if ghcr.io is reachable through a proxy, if so,
+    Apply expected proxy config to juju model.
+
+    Args:
+        ops_test (OpsTest): ops_test plugin
+
+    Returns:
+        true if proxy was configured.
+    """
+    assert ops_test.model, "Model must be present"
+    controller = await ops_test.model.get_controller()
+    controller_model = await controller.get_model("controller")
+    controller_unit = controller_model.applications["controller"].units[0]
+    http_proxy, site = "http://squid.internal:3128", "https://ghcr.io"
```

Comment:
If Github is down, this code will think that this is not run behind a proxy. Maybe add another site e.g. google to double-check in case Github is down (again).

Reply:
RIght... this isn't the STRONGEST of tests to determine if there's a proxy config necessary.  I thought about just having a pytest command line switch that says "apply proxy config" -- maybe that's a safer option than trying to TEST if a site is available from behind some proxy. 

WDYT?

Reply:
This sounds safer, indeed